File: generate-logo.js
const { createCanvas } = require('canvas');
const fs = require('fs');

function createLogo() {
    const size = 512;
    const canvas = createCanvas(size, size);
    const ctx = canvas.getContext('2d');

    // Background gradient (purple to pink)
    const bgGradient = ctx.createLinearGradient(0, 0, 0, size);
    bgGradient.addColorStop(0, '#667eea');
    bgGradient.addColorStop(0.5, '#764ba2');
    bgGradient.addColorStop(1, '#f093fb');
    ctx.fillStyle = bgGradient;
    ctx.fillRect(0, 0, size, size);

    // Glass layers
    const layers = [
        { x: 80, y: 80, size: 352, opacity: 0.25, blur: 15 },
        { x: 100, y: 100, size: 312, opacity: 0.20, blur: 10 },
        { x: 120, y: 120, size: 272, opacity: 0.15, blur: 8 }
    ];

    layers.forEach(layer => {
        ctx.save();
        ctx.globalAlpha = layer.opacity;

        const gradient = ctx.createRadialGradient(
            layer.x + layer.size/2, layer.y + layer.size/2, 0,
            layer.x + layer.size/2, layer.y + layer.size/2, layer.size/2
        );
        gradient.addColorStop(0, 'rgba(255, 255, 255, 1)');
        gradient.addColorStop(0.5, 'rgba(255, 255, 255, 0.5)');
        gradient.addColorStop(1, 'rgba(255, 255, 255, 0.1)');

        ctx.fillStyle = gradient;
        ctx.shadowBlur = layer.blur;
        ctx.shadowColor = 'rgba(0, 0, 0, 0.2)';

        // Rounded rectangle
        const radius = 35;
        ctx.beginPath();
        ctx.moveTo(layer.x + radius, layer.y);
        ctx.lineTo(layer.x + layer.size - radius, layer.y);
        ctx.quadraticCurveTo(layer.x + layer.size, layer.y, layer.x + layer.size, layer.y + radius);
        ctx.lineTo(layer.x + layer.size, layer.y + layer.size - radius);
        ctx.quadraticCurveTo(layer.x + layer.size, layer.y + layer.size, layer.x + layer.size - radius, layer.y + layer.size);
        ctx.lineTo(layer.x + radius, layer.y + layer.size);
        ctx.quadraticCurveTo(layer.x, layer.y + layer.size, layer.x, layer.y + layer.size - radius);
        ctx.lineTo(layer.x, layer.y + radius);
        ctx.quadraticCurveTo(layer.x, layer.y, layer.x + radius, layer.y);
        ctx.closePath();
        ctx.fill();

        ctx.restore();
    });

    // Code brackets symbol
    ctx.save();
    ctx.strokeStyle = 'rgba(255, 255, 255, 0.85)';
    ctx.lineWidth = 14;
    ctx.lineCap = 'round';
    ctx.lineJoin = 'round';
    ctx.shadowBlur = 10;
    ctx.shadowColor = 'rgba(0, 0, 0, 0.3)';

    const centerX = size / 2;
    const centerY = size / 2;
    const bracketSize = 50;

    // Left bracket <
    ctx.beginPath();
    ctx.moveTo(centerX - bracketSize + 10, centerY - 40);
    ctx.lineTo(centerX - bracketSize - 35, centerY);
    ctx.lineTo(centerX - bracketSize + 10, centerY + 40);
    ctx.stroke();

    // Right bracket >
    ctx.beginPath();
    ctx.moveTo(centerX + bracketSize - 10, centerY - 40);
    ctx.lineTo(centerX + bracketSize + 35, centerY);
    ctx.lineTo(centerX + bracketSize - 10, centerY + 40);
    ctx.stroke();

    // Slash /
    ctx.beginPath();
    ctx.moveTo(centerX - 10, centerY + 45);
    ctx.lineTo(centerX + 10, centerY - 45);
    ctx.stroke();

    ctx.restore();

    // Add sparkle effects
    const sparkles = [
        { x: 130, y: 130, size: 10 },
        { x: 390, y: 120, size: 8 },
        { x: 140, y: 380, size: 9 },
        { x: 380, y: 370, size: 11 },
        { x: 256, y: 100, size: 7 }
    ];

    sparkles.forEach(sparkle => {
        ctx.save();
        ctx.fillStyle = 'rgba(255, 255, 255, 0.6)';
        ctx.shadowBlur = 8;
        ctx.shadowColor = 'rgba(255, 255, 255, 0.8)';
        ctx.beginPath();
        ctx.arc(sparkle.x, sparkle.y, sparkle.size, 0, Math.PI * 2);
        ctx.fill();

        // Add star points
        ctx.fillStyle = 'rgba(255, 255, 255, 0.9)';
        ctx.shadowBlur = 4;
        for (let i = 0; i < 4; i++) {
            ctx.save();
            ctx.translate(sparkle.x, sparkle.y);
            ctx.rotate((Math.PI / 2) * i);
            ctx.beginPath();
            ctx.moveTo(0, -sparkle.size * 1.5);
            ctx.lineTo(sparkle.size * 0.3, 0);
            ctx.lineTo(0, sparkle.size * 0.4);
            ctx.lineTo(-sparkle.size * 0.3, 0);
            ctx.closePath();
            ctx.fill();
            ctx.restore();
        }
        ctx.restore();
    });

    // Save main icon
    const buffer = canvas.toBuffer('image/png');
    fs.writeFileSync('icon.png', buffer);
    console.log('✓ Logo created: icon.png (512x512)');

    // Create smaller versions
    const sizes = [
        { size: 128, name: 'icon-128.png' },
        { size: 64, name: 'icon-64.png' }
    ];

    sizes.forEach(({ size: targetSize, name }) => {
        const smallCanvas = createCanvas(targetSize, targetSize);
        const smallCtx = smallCanvas.getContext('2d');
        smallCtx.drawImage(canvas, 0, 0, targetSize, targetSize);
        const smallBuffer = smallCanvas.toBuffer('image/png');
        fs.writeFileSync(name, smallBuffer);
        console.log(`✓ Created: ${name} (${targetSize}x${targetSize})`);
    });

    console.log('\n🎨 All logos created successfully!');
}

createLogo();

File: generate_logo.py
#!/usr/bin/env python3
from PIL import Image, ImageDraw, ImageFilter, ImageFont
import math

def create_logo():
    # Create image
    size = 512
    img = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)
    
    # Background gradient (purple to pink)
    for y in range(size):
        r = int(102 + (240 - 102) * (y / size))
        g = int(126 + (147 - 126) * (y / size))
        b = int(234 + (251 - 234) * (y / size))
        draw.line([(0, y), (size, y)], fill=(r, g, b, 255))
    
    # Create glass layers
    glass_layer = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    glass_draw = ImageDraw.Draw(glass_layer)
    
    # Main glass rectangle
    margin = 80
    glass_rect = [margin, margin, size - margin, size - margin]
    radius = 40
    
    # Draw rounded rectangle with gradient effect
    for i in range(3):
        offset = i * 15
        alpha = int(80 - i * 20)
        glass_draw.rounded_rectangle(
            [margin + offset, margin + offset, size - margin - offset, size - margin - offset],
            radius=radius - i * 5,
            fill=(255, 255, 255, alpha)
        )
    
    # Apply blur for glass effect
    glass_layer = glass_layer.filter(ImageFilter.GaussianBlur(8))
    
    # Composite
    img = Image.alpha_composite(img, glass_layer)
    
    # Add VSCode-inspired icon overlay
    overlay = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    overlay_draw = ImageDraw.Draw(overlay)
    
    # Draw simplified brackets/code symbol
    center_x, center_y = size // 2, size // 2
    bracket_size = 60
    line_width = 12
    
    # Left bracket
    overlay_draw.line(
        [(center_x - bracket_size, center_y - 30),
         (center_x - bracket_size - 20, center_y - 30)],
        fill=(255, 255, 255, 200), width=line_width
    )
    overlay_draw.line(
        [(center_x - bracket_size - 20, center_y - 30),
         (center_x - bracket_size - 30, center_y)],
        fill=(255, 255, 255, 200), width=line_width
    )
    overlay_draw.line(
        [(center_x - bracket_size - 30, center_y),
         (center_x - bracket_size - 20, center_y + 30)],
        fill=(255, 255, 255, 200), width=line_width
    )
    overlay_draw.line(
        [(center_x - bracket_size - 20, center_y + 30),
         (center_x - bracket_size, center_y + 30)],
        fill=(255, 255, 255, 200), width=line_width
    )
    
    # Right bracket
    overlay_draw.line(
        [(center_x + bracket_size, center_y - 30),
         (center_x + bracket_size + 20, center_y - 30)],
        fill=(255, 255, 255, 200), width=line_width
    )
    overlay_draw.line(
        [(center_x + bracket_size + 20, center_y - 30),
         (center_x + bracket_size + 30, center_y)],
        fill=(255, 255, 255, 200), width=line_width
    )
    overlay_draw.line(
        [(center_x + bracket_size + 30, center_y),
         (center_x + bracket_size + 20, center_y + 30)],
        fill=(255, 255, 255, 200), width=line_width
    )
    overlay_draw.line(
        [(center_x + bracket_size + 20, center_y + 30),
         (center_x + bracket_size, center_y + 30)],
        fill=(255, 255, 255, 200), width=line_width
    )
    
    # Add sparkle effects
    sparkles = [
        (150, 150, 8), (380, 140, 6), (160, 370, 7),
        (370, 360, 9), (256, 120, 5), (256, 390, 6)
    ]
    
    for x, y, size in sparkles:
        overlay_draw.ellipse([x - size, y - size, x + size, y + size],
                           fill=(255, 255, 255, 150))
    
    # Composite overlay
    img = Image.alpha_composite(img, overlay)
    
    # Save
    img.save('/Users/qinyuanmao/vscode/vscode-liquid-glass/icon.png')
    print("Logo saved as icon.png")
    
    # Also create smaller versions for VSCode
    img_128 = img.resize((128, 128), Image.Resampling.LANCZOS)
    img_128.save('/Users/qinyuanmao/vscode/vscode-liquid-glass/icon-128.png')
    
    img_64 = img.resize((64, 64), Image.Resampling.LANCZOS)
    img_64.save('/Users/qinyuanmao/vscode/vscode-liquid-glass/icon-64.png')
    
    print("Additional sizes created: icon-128.png, icon-64.png")

if __name__ == '__main__':
    create_logo()
